doc/filters.md

```diff
@@ -56,14 +54,54 @@ This filter is intended to be used prior to the request being handled so that on
 
 ### Examples
 
-Post-authorization.
-
 ```tcl
 foreach http_method [list GET HEAD POST] {
     ns_register_filter postauth $http_method /* qc::filter_authenticate
 }
 ```
 
+
+qc::filter_http_request_validate
+-----------------------------
+
+The task of this filter is to check that the request and connection URL are both valid i.e. well formed.
+
+If either the request or the URL are invalid then a `400 Bad Request` is returned to the client with a message explaining which part was not valid.
+
+### Usage
+
+This filter is intended to be used prior to the request being handled and ideally before any other filters. Pre-authorization is the suggested time to use this filter so that invalid requests will be caught before anything else is attempted.
+
+### Examples
+
+```tcl
+foreach http_method [list GET POST HEAD] {
+    ns_register_filter preauth $http_method /* qc::filter_http_request_validate
+}
+```
+
+
+qc::filter_file_alias_paths
+---------------------------
+
+The task of this filter is to deal with requests that are an alias for a file. For example, a request for `/foo` might be an alias and the actual file path might be `/file/foo.png`.
+
+If the request URL is an alias for a file then the actual file path is found and a symbiotic link created if it doesn't exist before finally registering the alias on fastpath. If anything goes wrong the error is returned to the client.
+
```

Comment:
If the request URL is an alias for a file a symbolic link to the canonical file location is created (if it doesn't already exist) and the URL is registered for fastpath. If anything goes wrong the error is returned to the client.